src/ResourceStore.php

```diff
@@ -227,15 +229,24 @@ public function peek($type, $id)
 
     public function save(Resource $resource)
     {
-        $method = ($resource->getId() == '') ? 'POST' : 'PATCH';
+        $method = 'POST';
+        $json = $resource->encode();
+
+        if ($resource->getId() != '') {
+            $method = 'PATCH';
+
+            if ($this->disable_to_many_replace) {
+                $json = $this->removeToManyRelationships($json);
```

Comment:
rather than serializing and then modifying the serialized data, can we pass parameters to the encode methods so we get back an appropriate object?